packages/mc-scripts/src/commands/deployment-previews-push.ts

```diff
@@ -0,0 +1,266 @@
+export const fqdnRegex =
+  /(?=^.{4,253}$)(^((?!-)[a-zA-Z0-9-]{0,62}[a-zA-Z0-9]\.)+[a-zA-Z]{2,63}$)/i;
+
+const validateUrl = (url = '') => {
+  try {
+    const urlSchema = new URL(url);
+    const isProtocolValid = ['http:', 'https:'].includes(urlSchema.protocol);
+    const isHostnameValid = urlSchema.hostname.match(fqdnRegex);
+    return isProtocolValid && isHostnameValid;
+  } catch (error) {
+    return false;
+  }
+};
+
+const validateAlias = (alias = '') => {
+  return Boolean(alias.match(deploymentPreviewAliasRegex));
+};
+
+type TGetMcUrlLink = {
+  mcApiUrl: string;
+  organizationId: string;
+  customEntityId: string;
+  isCustomView?: boolean;
+};
+
+const getMcUrlLink = ({
+  mcApiUrl,
+  organizationId,
+  customEntityId,
+  isCustomView,
+}: TGetMcUrlLink) => {
+  const mcUrl = mcApiUrl.replace('mc-api', 'mc');
+  const customEntityLink = `${mcUrl}/account/organizations/${organizationId}/custom-${
+    isCustomView ? 'views' : 'applications'
+  }/owned/${customEntityId}/deployment-previews`;
+  return customEntityLink;
+};
+
+const isCustomViewData = (
+  data: CustomApplicationData | CustomViewData
+): data is CustomViewData =>
+  (data as CustomApplicationData).entryPointUriPath === undefined;
+
+type TPushDeploymentPreviewParams = {
+  mcApiUrl: string;
+  applicationId: string;
+  applicationIdentifier: string;
+  isCustomView: boolean;
+  localCustomEntityData: CustomApplicationData;
+  options: TCliCommandPushDeploymentPreviewOptions;
+};
+
+async function pushDeploymentPreview({
+  mcApiUrl,
+  applicationId,
+  applicationIdentifier,
+  localCustomEntityData,
+  isCustomView,
+  options,
+}: TPushDeploymentPreviewParams) {
+  /*
+    1. Check if the Custom Application exists in the Merchant Center.
+    2. Ask for the alias and URL of the deployment preview.
+    3. Check whether it exists a deployment preview with the same alias.
+    4. If it exists, ask for confirmation to update it.
+    5. If it doesn't exist, ask for confirmation to create it.
+  */
+
+  const fetchedCustomApplication = await fetchCustomApplication({
+    mcApiUrl,
+    entryPointUriPath: localCustomEntityData.entryPointUriPath,
+    applicationIdentifier,
+  });
+
+  if (!fetchedCustomApplication) {
+    throw new Error(
+      `The Custom Application "${applicationIdentifier}" does not exist in the Merchant Center.`
+    );
+  }
+
+  let deploymentAlias = options.alias;
+  // If the alias was not provided as an argument to the script, let's ask for it.
+  if (!deploymentAlias) {
+    const { alias } = await prompts({
+      type: 'text',
+      name: 'alias',
+      message: [
+        `What's the alias for the deployment preview? (my-alias)`,
+        options.dryRun &&
+          chalk.gray(
+            'Using "--dry-run", no deployment preview will be created.'
+          ),
+      ]
+        .filter(Boolean)
+        .join('\n'),
+      validate: (value) => value && value.length > 2,
```

Comment:
Nit: should we suggest a default alias like `staging` or something?
I would also rephrase a bit the sentence:

```suggestion
    const { alias } = await prompts({
      type: 'text',
      name: 'alias',
      initial: 'my-alias',
      message: [
        `Enter the alias for the deployment preview`,
        options.dryRun &&
          chalk.gray(
            'Using "--dry-run", no deployment preview will be created.'
          ),
      ]
        .filter(Boolean)
        .join('\n'),
      validate: (value) => value && value.length > 2,
```

Reply:
I updated the message here: 54c523145d410c538431c6c27eca10866e794a0b

Regarding the default value, I don't see the point as any consumer could use something different.